src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -621,10 +700,21 @@ protected override void Dispose(bool disposing)
                 public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
                 {
                     Http2Stream http2Stream = _http2Stream;
-                    Task t = http2Stream != null ?
-                        http2Stream.SendDataAsync(buffer, cancellationToken) :
-                        Task.FromException(new ObjectDisposedException(nameof(Http2WriteStream)));
-                    return new ValueTask(t);
+
+                    if (http2Stream == null)
+                    {
+                        return new ValueTask(Task.FromException(new ObjectDisposedException(nameof(Http2WriteStream))));
+                    }
+
+                    // We got rejected while sending request.
+                    if (http2Stream._response != null && (int)http2Stream._response.StatusCode >= 300)
```

Comment:
Is it possible the response could get nulled out concurrently?  Hopefully not.  If it could, it should be stored into a local and then the local used for both the check and access.

Reply:
no, it should not. We never set it to null once allocated.

Reply:
It does seem kind of weird that we are explicitly checking the status code here.

How is this handled in the HTTP/1.1 case?